Docs-4013-DH

dhirsch1 opened the pull request:
This request is for updates to four topics: 

- Setting Environment Variables
- Notifications Configuration Reference
- Reproducible Builds
- The TOC (to include the above three topics).

The topics were migrated as is, except that I removed links that did not point to 4.1 topics and also tweaked the wording, so they need to be reviewed.

Commit:
Adding new files
Adding new files

Commit:
New files added
New fkiles added

Commit:
Work on DOCS-4013
Work on DOCS-4013

Commit:
Added one file and lnk
Added one fi;e and link

Commit:
Committing a straggler file
Commmitting a straggler file

Commit:
Added another log file
Added another log file

Commit:
Fixed DOCS-3285 and moved TLS topic under Security
Fixed DOCS-3285 and moved TLS topic under Security

Commit:
Copyedit of the file
Copyedit of the file

Commit:
More editing changes
More editing changes

Commit:
Added missing graphics files
Added missing graphics files

Commit:
More writing tweaks
More writing tweaksX

dhirsch1 committed:
[committing to split out the logging topic from the branch]

dhirsch1 committed:
Copyedit of the notifications_reference topic
Copyedit of the notifications_reference topic

dhirsch1 committed:
Updates to two topics and a new graphic
Updates to two topics and a new graphjic

dhirsch1 committed:
Updates to topics and TOC
Updates to topics and TOC

dhirsch1 commented:
Alejandro has reviewed this so it is waiting on Pablo to review the Notifications Configuration Reference, which he said he can do next week.

pablolagreca commented on modules/ROOT/pages/logging.adoc:
```diff
@@ -0,0 +1,93 @@
+
+If you need more details about the message, a simple scripted logging component like the following can come handy:
+
+[source,xml,linenums]
```
I would remove this. We don't want to promote this in Mule 4

dhirsch1 replied:
This will be in another pull request. I will apply your comment then.

pablolagreca commented on modules/ROOT/pages/notifications-configuration-reference.adoc:
```diff
@@ -0,0 +1,119 @@
+You can specify the following types of notifications using the `event` attribute of the `notification` and `disable-notification` elements:
+
+* ASYNC-MESSAGE
+* COMPONENT-MESSAGE
```
Does not exists in Mule 4

dhirsch1 replied:
Removed

pablolagreca commented on modules/ROOT/pages/notifications-configuration-reference.adoc:
```diff
@@ -0,0 +1,119 @@
+* CONNECTION
+* CONTEXT
+* CUSTOM
+* ENDPOINT-MESSAGE
```
Does not exists in Mule 4

dhirsch1 replied:
Removed

pablolagreca commented on modules/ROOT/pages/notifications-configuration-reference.adoc:
```diff
@@ -0,0 +1,119 @@
+* CUSTOM
+* ENDPOINT-MESSAGE
+* EXCEPTION
+* EXCEPTION-STRATEGY
```
Does not exists in Mule 4

dhirsch1 replied:
Removed.

dhirsch1 committed:
Added Pablo's comments on notification events
Added Pablo's comments on notification events

dhirsch1 committed:
Removed logging and debug files that are not ready. They will be in another pull request. Fixed TOC to reflect this.
Removed logging and debug files that are not ready. They will be in another pull request. Fixed TOC to reflect this.    Y

martincousido commented on modules/ROOT/pages/setting-environment-variables.adoc:
```diff
@@ -0,0 +1,24 @@
+= Setting Environment Variables
```
Should we add something about the env variables for the mule that we use for tooling? Some people might think that adding those env variables to the run configuration would also affect the design time of the mule application, which won't be the case. @pablolagreca What do you think?

pablolagreca replied:
Yes. Make sense. Please provide the content to be added to Diane

aiannucci commented on modules/ROOT/pages/notifications-configuration-reference.adoc:
```diff
@@ -0,0 +1,116 @@
+|===
+|Name |Description
+|`dynamic` |If the notification manager is dynamic, you can set up the listeners to be registered dynamically at runtime. Otherwise, some parts of Mule will cache the notification configurations for efficiency reasons and will not generate events for newly-enabled notifications or listeners.
+//See https://docs.mulesoft.com/mule-runtime/3.9/mule-server-notifications#registering-listeners-programmatically for more information.//
```
Is this linking to 3.9 docs?

dhirsch1 replied:
It is commented out. I will remove the comments anyway.

dhirsch1 commented:
This pull request had some topics split out to be done later. See PR 4013-DH1
